django/contrib/auth/__init__.py

```diff
@@ -108,6 +110,13 @@ def login(request, user, backend=None):
     session_auth_hash = ""
     if user is None:
         user = request.user
+        if user and isinstance(user, get_user_model()):
```

Comment:
We would need to add some extra commentary around the code that is being deprecated. The specifics are described in [these docs](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#deprecating-a-feature).